opennebula/resource_opennebula_virtual_network.go

```diff
@@ -396,16 +414,17 @@ func resourceOpennebulaVirtualNetworkCreate(d *schema.ResourceData, meta interfa
 		}
 
 		//The API only takes ATTRIBUTE=VALUE for VNET reservations...
-		reservation_string := "SIZE=%d\nNAME=\"%s\""
+		reservation_string := "SIZE=%d\nNAME=\"%s\"\nAR_ID=%d\nIP=\"%s\""
```

Comment:
From the documentation, only `SIZE` is mandatory, other fields are optional.

By the way, I know that you adapted existing code, it's fine, but we could improve this a bit.
In Goca, there is some facilities to build dynamic template parts, so instead of using a format string we could  build it like this:
```
       import "github.com/OpenNebula/one/src/oca/go/src/goca/dynamic"
       
       ...
       
	tpl := dynamic.NewTemplate()
	tpl.AddPair("NAME", "thisisatest")
	tpl.AddPair("SIZE", 1)

	vm, err := controller.VirtualNetwork(vnID).Reserve(tpl.String())
```
In that way, it would be easy to test if an attribute is set via `Get` or `GetOk`, and if so, put the corresponding element in the template

Reply:
Excellent! This is much tidier, I will do that, also.